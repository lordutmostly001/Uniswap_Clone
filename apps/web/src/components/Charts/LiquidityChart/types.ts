import { CustomData, CustomSeriesOptions, UTCTimestamp } from 'lightweight-charts'

export interface LiquidityBarData extends CustomData {
  time: UTCTimestamp
  tick: number
  price0: string
  price1: string
  liquidity: number
  amount0Locked: number
  amount1Locked: number
}

export interface LiquidityBarProps {
  tokenAColor: string
  tokenBColor: string
  highlightColor: string
  activeTick?: number
  activeTickProgress?: number
}

export interface LiquidityBarSeriesOptions extends CustomSeriesOptions, LiquidityBarProps {
  hoveredTick?: number
}
