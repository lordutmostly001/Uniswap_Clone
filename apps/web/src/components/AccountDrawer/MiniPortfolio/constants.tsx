import type { BaseTransactionType } from 'state/transactions/types'
import { UniswapXOrderStatus } from 'types/uniswapx'
import {
  TransactionStatus,
  TransactionType as UniswapTransactionType,
} from 'uniswap/src/features/transactions/types/transactionDetails'
import i18n from 'uniswap/src/i18n'
import { logger } from 'utilities/src/logger/logger'

// use even number because rows are in groups of 2
export const DEFAULT_NFT_QUERY_AMOUNT = 26

type TransactionStatusWeb = TransactionStatus.Success | TransactionStatus.Failed | TransactionStatus.Pending

const getTransactionTitleTable = (): {
  [key in BaseTransactionType]: {
    [state in TransactionStatusWeb]: string
  }
} => ({
  [UniswapTransactionType.Swap]: {
    [TransactionStatus.Pending]: i18n.t('common.swapping'),
    [TransactionStatus.Success]: i18n.t('common.swapped'),
    [TransactionStatus.Failed]: i18n.t('common.swap.failed'),
  },
  [UniswapTransactionType.Wrap]: {
    [TransactionStatus.Pending]: i18n.t('common.wrapping'),
    [TransactionStatus.Success]: i18n.t('common.wrapped'),
    [TransactionStatus.Failed]: i18n.t('common.wrap.failed'),
  },
  [UniswapTransactionType.CollectFees]: {
    [TransactionStatus.Pending]: i18n.t('common.collecting.fees'),
    [TransactionStatus.Success]: i18n.t('common.collected.fees'),
    [TransactionStatus.Failed]: i18n.t('common.collect.fees.failed'),
  },
  [UniswapTransactionType.LPIncentivesClaimRewards]: {
    [TransactionStatus.Pending]: i18n.t('pool.incentives.collectingRewards'),
    [TransactionStatus.Success]: i18n.t('pool.incentives.collectedRewards'),
    [TransactionStatus.Failed]: i18n.t('pool.incentives.collectFailedNoRetry'),
  },
  [UniswapTransactionType.Approve]: {
    [TransactionStatus.Pending]: i18n.t('common.approving'),
    [TransactionStatus.Success]: i18n.t('common.approved'),
    [TransactionStatus.Failed]: i18n.t('common.approval.failed'),
  },
  [UniswapTransactionType.ClaimUni]: {
    [TransactionStatus.Pending]: i18n.t('common.claiming'),
    [TransactionStatus.Success]: i18n.t('common.claimed'),
    [TransactionStatus.Failed]: i18n.t('common.claim.failed'),
  },
  [UniswapTransactionType.Send]: {
    [TransactionStatus.Pending]: i18n.t('common.sending'),
    [TransactionStatus.Success]: i18n.t('common.sent'),
    [TransactionStatus.Failed]: i18n.t('common.send.failed'),
  },
  [UniswapTransactionType.MigrateLiquidityV2ToV3]: {
    [TransactionStatus.Pending]: i18n.t('common.migrating.liquidity'),
    [TransactionStatus.Success]: i18n.t('common.migrated.liquidity'),
    [TransactionStatus.Failed]: i18n.t('common.migrate.liquidity.failed'),
  },
  [UniswapTransactionType.LiquidityIncrease]: {
    [TransactionStatus.Pending]: i18n.t('common.adding.liquidity'),
    [TransactionStatus.Success]: i18n.t('common.added.liquidity'),
    [TransactionStatus.Failed]: i18n.t('common.add.liquidity.failed'),
  },
  [UniswapTransactionType.LiquidityDecrease]: {
    [TransactionStatus.Pending]: i18n.t('common.removing.liquidity'),
    [TransactionStatus.Success]: i18n.t('common.liquidity.removed'),
    [TransactionStatus.Failed]: i18n.t('common.remove.liquidity.failed'),
  },
  [UniswapTransactionType.CreatePool]: {
    [TransactionStatus.Pending]: i18n.t('position.create.modal.header'),
    [TransactionStatus.Success]: i18n.t('pool.createdPosition'),
    [TransactionStatus.Failed]: i18n.t('pool.createdPosition.failed'),
  },
  [UniswapTransactionType.CreatePair]: {
    [TransactionStatus.Pending]: i18n.t('position.create.modal.header'),
    [TransactionStatus.Success]: i18n.t('pool.createdPosition'),
    [TransactionStatus.Failed]: i18n.t('pool.createdPosition.failed'),
  },
  [UniswapTransactionType.MigrateLiquidityV3ToV4]: {
    [TransactionStatus.Pending]: i18n.t('common.migrating.liquidity'),
    [TransactionStatus.Success]: i18n.t('common.migrated.liquidity'),
    [TransactionStatus.Failed]: i18n.t('common.migrate.liquidity.failed'),
  },
  [UniswapTransactionType.Bridge]: {
    [TransactionStatus.Pending]: i18n.t('common.swapping'),
    [TransactionStatus.Success]: i18n.t('common.swapped'),
    [TransactionStatus.Failed]: i18n.t('common.swap.failed'),
  },
  [UniswapTransactionType.Permit2Approve]: {
    [TransactionStatus.Pending]: i18n.t('common.approving'),
    [TransactionStatus.Success]: i18n.t('common.permit.approved'),
    [TransactionStatus.Failed]: i18n.t('common.permit.failed'),
  },
})

export const getCancelledTransactionTitleTable = (): { [key in BaseTransactionType]: string } => ({
  [UniswapTransactionType.Swap]: i18n.t('common.swap.cancelled'),
  [UniswapTransactionType.Wrap]: i18n.t('common.wrap.cancelled'),
  [UniswapTransactionType.CollectFees]: i18n.t('common.collect.fees.cancelled'),
  [UniswapTransactionType.Approve]: i18n.t('common.approval.cancelled'),
  [UniswapTransactionType.ClaimUni]: i18n.t('common.claim.cancelled'),
  [UniswapTransactionType.LPIncentivesClaimRewards]: i18n.t('pool.incentives.collectRewardsCancelled'),
  [UniswapTransactionType.Send]: i18n.t('common.send.cancelled'),
  [UniswapTransactionType.MigrateLiquidityV2ToV3]: i18n.t('common.migrate.liquidity.cancelled'),
  [UniswapTransactionType.LiquidityIncrease]: i18n.t('common.add.liquidity.cancelled'),
  [UniswapTransactionType.LiquidityDecrease]: i18n.t('common.remove.liquidity.cancelled'),
  [UniswapTransactionType.CreatePool]: i18n.t('pool.createdPosition.cancelled'),
  [UniswapTransactionType.CreatePair]: i18n.t('pool.createdPosition.cancelled'),
  [UniswapTransactionType.MigrateLiquidityV3ToV4]: i18n.t('common.migrate.liquidity.cancelled'),
  [UniswapTransactionType.Bridge]: i18n.t('common.swap.cancelled'),
  [UniswapTransactionType.Permit2Approve]: i18n.t('common.permit.cancelled'),
})

const getAlternateTransactionTitleTable = (): {
  [key in BaseTransactionType]?: { [state in TransactionStatusWeb]: string }
} => ({
  [UniswapTransactionType.Wrap]: {
    [TransactionStatus.Pending]: i18n.t('common.unwrapping'),
    [TransactionStatus.Success]: i18n.t('common.unwrapped'),
    [TransactionStatus.Failed]: i18n.t('common.unwrap.failed'),
  },
  [UniswapTransactionType.Approve]: {
    [TransactionStatus.Pending]: i18n.t('common.revoking.approval'),
    [TransactionStatus.Success]: i18n.t('common.revoked.approval'),
    [TransactionStatus.Failed]: i18n.t('common.revoke.approval.failed'),
  },
})

export function getActivityTitle({
  type,
  status,
  alternate,
}: {
  type: BaseTransactionType
  status: TransactionStatus
  alternate?: boolean
}): string {
  if (
    status !== TransactionStatus.Pending &&
    status !== TransactionStatus.Failed &&
    status !== TransactionStatus.Success
  ) {
    logger.error(new Error(`Unhandled web transaction status`), {
      tags: {
        file: 'constants.tsx',
        function: 'getActivityTitle',
      },
      extra: { status },
    })
    return ''
  }

  if (alternate) {
    const AlternateTransactionTitleTable = getAlternateTransactionTitleTable()
    const alternateTitle = AlternateTransactionTitleTable[type]
    if (alternateTitle !== undefined) {
      return alternateTitle[status]
    }
  }

  const TransactionTitleTable = getTransactionTitleTable()
  return TransactionTitleTable[type][status]
}

interface OrderTextTableEntry {
  status: TransactionStatus
  getTitle: () => string
  getStatusMessage?: () => string
}

export const getOrderTextTable = (): {
  [status in UniswapXOrderStatus]: OrderTextTableEntry
} => {
  const TransactionTitleTable = getTransactionTitleTable()
  const SwapTitleTable = TransactionTitleTable[UniswapTransactionType.Swap]

  return {
    [UniswapXOrderStatus.OPEN]: {
      getTitle: () => SwapTitleTable[TransactionStatus.Pending],
      status: TransactionStatus.Pending,
    },
    [UniswapXOrderStatus.FILLED]: {
      getTitle: () => SwapTitleTable[TransactionStatus.Success],
      status: TransactionStatus.Success,
    },
    [UniswapXOrderStatus.EXPIRED]: {
      getTitle: () => i18n.t('common.swap.expired'),
      getStatusMessage: () => i18n.t('common.your.swap.could.not.be.fulfilled'),
      status: TransactionStatus.Failed,
    },
    [UniswapXOrderStatus.ERROR]: {
      getTitle: () => SwapTitleTable[TransactionStatus.Failed],
      status: TransactionStatus.Failed,
    },
    [UniswapXOrderStatus.INSUFFICIENT_FUNDS]: {
      getTitle: () => i18n.t('common.insufficient.funds'),
      getStatusMessage: () => i18n.t('common.your.account.had.insufficient.funds'),
      status: TransactionStatus.Failed,
    },
    [UniswapXOrderStatus.PENDING_CANCELLATION]: {
      getTitle: () => i18n.t('common.pending.cancellation'),
      status: TransactionStatus.Pending,
    },
    [UniswapXOrderStatus.CANCELLED]: {
      getTitle: () => i18n.t('common.swap.cancelled'),
      status: TransactionStatus.Failed,
    },
  }
}

export const getLimitOrderTextTable = (): {
  [status in UniswapXOrderStatus]: OrderTextTableEntry
} => ({
  [UniswapXOrderStatus.OPEN]: {
    getTitle: () => i18n.t('common.limit.opened'),
    status: TransactionStatus.Pending,
  },
  [UniswapXOrderStatus.FILLED]: {
    getTitle: () => i18n.t('common.limit.executed'),
    status: TransactionStatus.Success,
  },
  [UniswapXOrderStatus.EXPIRED]: {
    getTitle: () => i18n.t('common.limit.expired'),
    getStatusMessage: () => i18n.t('common.your.limit.could.not.be.fulfilled'),
    status: TransactionStatus.Failed,
  },
  [UniswapXOrderStatus.ERROR]: {
    getTitle: () => i18n.t('common.limit.failed'),
    status: TransactionStatus.Failed,
  },
  [UniswapXOrderStatus.INSUFFICIENT_FUNDS]: {
    getTitle: () => i18n.t('common.limit.opened'),
    getStatusMessage: () => i18n.t('common.your.account.has.insufficient.funds'),
    status: TransactionStatus.Pending,
  },
  [UniswapXOrderStatus.PENDING_CANCELLATION]: {
    getTitle: () => i18n.t('common.pending.cancellation'),
    status: TransactionStatus.Pending,
  },
  [UniswapXOrderStatus.CANCELLED]: {
    getTitle: () => i18n.t('common.limit.cancelled'),
    status: TransactionStatus.Failed,
  },
})

// Non-exhaustive list of addresses Moonpay uses when sending purchased tokens
export const MOONPAY_SENDER_ADDRESSES = [
  '0x8216874887415e2650d12d53ff53516f04a74fd7',
  '0x151b381058f91cf871e7ea1ee83c45326f61e96d',
  '0xb287eac48ab21c5fb1d3723830d60b4c797555b0',
  '0xd108fd0e8c8e71552a167e7a44ff1d345d233ba6',
]
