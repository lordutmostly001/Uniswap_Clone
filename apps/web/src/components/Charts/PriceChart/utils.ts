import { PricePoint } from 'appGraphql/data/util'
import { PriceChartData } from 'components/Charts/PriceChart'
import { CandlestickData } from 'lightweight-charts'

/**
 * Returns the minimum and maximum values in the given array of PricePoints.
 */
export function getPriceBounds(prices: PricePoint[]): { min: number; max: number } {
  if (!prices.length) {
    return { min: 0, max: 0 }
  }

  let min = prices[0].value
  let max = prices[0].value

  for (const pricePoint of prices) {
    if (pricePoint.value < min) {
      min = pricePoint.value
    }
    if (pricePoint.value > max) {
      max = pricePoint.value
    }
  }

  return { min, max }
}

/**
 * Returns the minimum and maximum values in the given array of candlestick data.
 */
export function getCandlestickPriceBounds(data: CandlestickData[]): { min: number; max: number } {
  if (!data.length) {
    return { min: 0, max: 0 }
  }

  let min = data[0].low
  let max = data[0].high

  for (const dataPoint of data) {
    if (dataPoint.low < min) {
      min = dataPoint.low
    }
    if (dataPoint.high > max) {
      max = dataPoint.high
    }
  }

  return { min, max }
}

/**
 * Returns the currentPrice or the most recent price from the data array.
 */
export function getEffectivePrice({
  currentPrice,
  data,
}: {
  currentPrice: number | undefined
  data: PriceChartData[]
}): number {
  if (currentPrice) {
    return currentPrice
  }
  return data[data.length - 1]?.value ?? 0
}
