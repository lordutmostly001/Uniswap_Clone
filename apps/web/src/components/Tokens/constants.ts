// Breakpoints specifically for the token pages

// TODO(WEB-2968): Deprecate these in the new .info project
export const MAX_WIDTH_MEDIA_BREAKPOINT = '1200px'
