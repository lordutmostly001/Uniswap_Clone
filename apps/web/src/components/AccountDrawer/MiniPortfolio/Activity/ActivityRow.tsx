import { useOpenOffchainActivityModal } from 'components/AccountDrawer/MiniPortfolio/Activity/OffchainActivityModal'
import { useTimeSince } from 'components/AccountDrawer/MiniPortfolio/Activity/parseRemote'
import { Activity } from 'components/AccountDrawer/MiniPortfolio/Activity/types'
import { PortfolioLogo } from 'components/AccountDrawer/MiniPortfolio/PortfolioLogo'
import PortfolioRow from 'components/AccountDrawer/MiniPortfolio/PortfolioRow'
import AlertTriangleFilled from 'components/Icons/AlertTriangleFilled'
import { LoaderV2 } from 'components/Icons/LoadingSpinner'
import Column from 'components/deprecated/Column'
import Row from 'components/deprecated/Row'
import styled from 'lib/styled-components'
import { useCallback } from 'react'
import { SignatureType } from 'state/signatures/types'
import { ThemedText } from 'theme/components'
import { EllipsisStyle } from 'theme/components/styles'
import { BridgeIcon } from 'uniswap/src/components/CurrencyLogo/SplitLogo'
import { TransactionType } from 'uniswap/src/data/graphql/uniswap-data-api/__generated__/types-and-hooks'
import Trace from 'uniswap/src/features/telemetry/Trace'
import { ElementName } from 'uniswap/src/features/telemetry/constants'
import { TransactionStatus } from 'uniswap/src/features/transactions/types/transactionDetails'
import { ExplorerDataType, getExplorerLink } from 'uniswap/src/utils/linking'
import { logger } from 'utilities/src/logger/logger'
import { isHash } from 'viem'

const ActivityRowDescriptor = styled(ThemedText.BodySmall)`
  color: ${({ theme }) => theme.neutral2};
  ${EllipsisStyle}
  max-width: 100%;
`

const StyledTimestamp = styled(ThemedText.BodySmall)`
  color: ${({ theme }) => theme.neutral2};
  line-height: 24px;
  font-variant: small;
  padding-right: 8px;
  font-feature-settings:
    'tnum' on,
    'lnum' on,
    'ss02' on;
`

function StatusIndicator({
  activity: { status, timestamp, offchainOrderDetails },
}: {
  activity: Activity
}): JSX.Element | null {
  const timeSince = useTimeSince(timestamp)

  switch (status) {
    case TransactionStatus.Pending:
      if (offchainOrderDetails?.type === SignatureType.SIGN_LIMIT) {
        return null
      }
      return <LoaderV2 />
    case TransactionStatus.Success:
      return <StyledTimestamp>{timeSince}</StyledTimestamp>
    case TransactionStatus.Failed:
      return <AlertTriangleFilled />
    default:
      logger.error(new Error(`Unhandled web transaction status`), {
        tags: {
          file: 'ActivityRow.tsx',
          function: 'StatusIndicator',
        },
        extra: { status },
      })
      return null
  }
}

// TODO WEB-4550 - Fix regression where ENS name is not displayed in activity row
export function ActivityRow({ activity }: { activity: Activity }) {
  const {
    chainId,
    title,
    descriptor,
    otherAccount,
    currencies,
    hash,
    prefixIconSrc,
    suffixIconSrc,
    offchainOrderDetails,
    logos,
    type,
  } = activity

  // TODO(WEB-5146): Create tamagui universal Activity component, remove one off bridge styling
  const isBridge = type === TransactionType.Bridging

  const openOffchainActivityModal = useOpenOffchainActivityModal()

  const explorerUrl = getExplorerLink({ chainId, data: hash, type: ExplorerDataType.TRANSACTION })

  const onClick = useCallback(() => {
    if (offchainOrderDetails) {
      openOffchainActivityModal(offchainOrderDetails, {
        inputLogo: activity.logos?.[0],
        outputLogo: activity.logos?.[1],
      })
      return
    }
    // Do not allow FOR activity to be opened until confirmed on chain
    if (activity.status === TransactionStatus.Pending && !isHash(hash)) {
      return
    }

    window.open(explorerUrl, '_blank')
  }, [activity.logos, activity.status, explorerUrl, hash, offchainOrderDetails, openOffchainActivityModal])

  return (
    <Trace
      logPress
      element={ElementName.MiniPortfolioActivityRow}
      properties={{ hash, chain_id: chainId, explorer_url: explorerUrl }}
    >
      <PortfolioRow
        left={
          <Column>
            <PortfolioLogo
              chainId={chainId}
              currencies={currencies}
              images={logos}
              accountAddress={otherAccount}
              customIcon={isBridge ? BridgeIcon : undefined}
            />
          </Column>
        }
        title={
          <Row align="space-between" justify-content="center">
            <Row gap="4px">
              {prefixIconSrc && <img height="14px" width="14px" src={prefixIconSrc} alt="" />}
              <ThemedText.SubHeader>{title}</ThemedText.SubHeader>
              {suffixIconSrc && <img height="14px" width="14px" src={suffixIconSrc} alt="" />}
            </Row>
            <StatusIndicator activity={activity} />
          </Row>
        }
        descriptor={<ActivityRowDescriptor color="neutral2">{descriptor}</ActivityRowDescriptor>}
        onClick={onClick}
      />
    </Trace>
  )
}
