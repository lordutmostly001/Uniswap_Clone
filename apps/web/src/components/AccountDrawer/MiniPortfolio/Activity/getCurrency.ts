import { Currency } from '@uniswap/sdk-core'
import { apolloClient } from 'appGraphql/data/apollo/client'
import { gqlTokenToCurrencyInfo } from 'appGraphql/data/types'
import { COMMON_BASES } from 'uniswap/src/constants/routing'
import { nativeOnChain } from 'uniswap/src/constants/tokens'
import {
  Token,
  TokenDocument,
  TokenQuery,
} from 'uniswap/src/data/graphql/uniswap-data-api/__generated__/types-and-hooks'
import { toGraphQLChain } from 'uniswap/src/features/chains/utils'
import { CurrencyId } from 'uniswap/src/types/currency'
import { currencyIdToAddress, currencyIdToChain, isNativeCurrencyAddress } from 'uniswap/src/utils/currencyId'
import { isSameAddress } from 'utilities/src/addresses'

export async function getCurrencyFromCurrencyId(currencyId: CurrencyId): Promise<Currency | undefined> {
  // Split currencyId and confirm validity
  const chainId = currencyIdToChain(currencyId)
  const address = currencyIdToAddress(currencyId)
  if (!chainId) {
    return undefined
  }

  // Handle native currency
  const isNative = isNativeCurrencyAddress(chainId, address)
  if (isNative) {
    return nativeOnChain(chainId)
  }

  // Handle common bases
  const commonBase = COMMON_BASES[chainId].find(
    (base) => base.currency.isToken && isSameAddress(base.currency.address, address),
  )
  if (commonBase) {
    return commonBase.currency
  }

  // Query for token from graphql
  const { data } = await apolloClient.query<TokenQuery>({
    query: TokenDocument,
    variables: {
      address,
      chain: toGraphQLChain(chainId),
    },
  })
  return gqlTokenToCurrencyInfo(data.token as Token)?.currency
}
