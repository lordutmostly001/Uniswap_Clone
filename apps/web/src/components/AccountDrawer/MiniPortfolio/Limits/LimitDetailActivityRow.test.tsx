import 'test-utils/tokens/mocks'

import { WETH9 } from '@uniswap/sdk-core'
import { Activity } from 'components/AccountDrawer/MiniPortfolio/Activity/types'
import { LimitDetailActivityRow } from 'components/AccountDrawer/MiniPortfolio/Limits/LimitDetailActivityRow'
import { SignatureType, UniswapXOrderDetails } from 'state/signatures/types'
import { render, screen } from 'test-utils/render'
import { UniswapXOrderStatus } from 'types/uniswapx'
import { DAI } from 'uniswap/src/constants/tokens'
import { UniverseChainId } from 'uniswap/src/features/chains/types'
import { TransactionStatus, TransactionType } from 'uniswap/src/features/transactions/types/transactionDetails'
import { currencyId } from 'uniswap/src/utils/currencyId'

vi.mock('components/AccountDrawer/MiniPortfolio/formatTimestamp', async () => {
  const actual = await vi.importActual('components/AccountDrawer/MiniPortfolio/formatTimestamp')
  return {
    ...actual,
    formatTimestamp: () => 'Expires January 1, 1970 at 12:00 AM',
  }
})

const mockOrderDetails: UniswapXOrderDetails = {
  type: SignatureType.SIGN_LIMIT,
  orderHash: '0x1234',
  status: UniswapXOrderStatus.OPEN,
  swapInfo: {
    isUniswapXOrder: true,
    type: TransactionType.Swap,
    tradeType: 0,
    inputCurrencyId: currencyId(DAI),
    outputCurrencyId: currencyId(WETH9[UniverseChainId.Mainnet]),
    inputCurrencyAmountRaw: '252074033564766400000',
    expectedOutputCurrencyAmountRaw: '106841079134757921',
    minimumOutputCurrencyAmountRaw: '106841079134757921',
    settledOutputCurrencyAmountRaw: '106841079134757921',
  },
  encodedOrder: '0xencodedOrder',
  id: '0x1234',
  addedTime: 3,
  chainId: UniverseChainId.Mainnet,
  expiry: 4,
  offerer: '0x1234',
}

const mockOrder: Activity = {
  hash: '0x123',
  chainId: UniverseChainId.Mainnet,
  status: TransactionStatus.Pending,
  timestamp: 1,
  title: 'Limit pending',
  from: '0x456',
  offchainOrderDetails: mockOrderDetails,
  currencies: [DAI, WETH9[UniverseChainId.Mainnet]],
}

describe('LimitDetailActivityRow', () => {
  it('should not render with no offchain order details', () => {
    const { container } = render(
      <LimitDetailActivityRow
        order={{ ...mockOrder, offchainOrderDetails: undefined }}
        onToggleSelect={vi.fn()}
        selected={false}
      />,
    )
    expect(container.firstChild?.firstChild?.firstChild).toBeNull()
  })

  it('should render with valid details', () => {
    // Addresses a console.error -- `Warning: React does not recognize the `scaleIcon` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `scaleicon` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
    // This is from tamagui's Checkbox component`
    vi.spyOn(console, 'error').mockImplementation(() => {})

    const { container } = render(<LimitDetailActivityRow onToggleSelect={vi.fn()} selected={false} order={mockOrder} />)
    expect(container.firstChild).toMatchSnapshot()
    expect(screen.getByText('when 0.00042 WETH/DAI')).toBeInTheDocument()
  })
})
