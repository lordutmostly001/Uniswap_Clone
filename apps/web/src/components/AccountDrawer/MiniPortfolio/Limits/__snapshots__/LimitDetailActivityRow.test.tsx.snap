// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LimitDetailActivityRow > should render with valid details 1`] = `
<span
  class=""
  style="display: contents;"
>
  <div
    class="_display-flex _flexBasis-auto _boxSizing-border-box _position-relative _minHeight-0px _minWidth-0px _flexShrink-0 _flexDirection-row _alignItems-center"
  >
    <div
      class="_whiteSpace-_platformweb_nowrap _overflowX-_platformweb_hidden _overflowY-_platformweb_hidden _textOverflow-_platformweb_ellipsis _display-flex _flexBasis-auto _boxSizing-border-box _position-relative _minHeight-0px _minWidth-0px _flexShrink-0 _flexDirection-row _gap-t-space-gap1569 _flexGrow-1 _pr-0px _pl-0px _cursor-pointer _alignItems-center _justifyContent-space-betwe3241 portfolio-row-wrapper undefined"
    >
      <div
        class="_display-flex _flexBasis-auto _boxSizing-border-box _position-relative _minHeight-0px _minWidth-0px _flexShrink-1 _flexDirection-column _alignItems-flex-start _flexGrow-1 _overflowX-hidden _overflowY-hidden"
      >
        <span
          class="font_body _display-inline _boxSizing-border-box _whiteSpace-pre-wrap _mt-0px _mr-0px _mb-0px _ml-0px _color-neutral2 _fontFamily-f-family _wordWrap-break-word _fontSize-f-size-micr111 _lineHeight-f-lineHeigh1263414315 _fontWeight-f-weight-bo3548"
          data-disable-theme="true"
        >
          Expires Expires January 1, 1970 at 12:00 AM
        </span>
        <div
          class="_display-flex _alignItems-stretch _flexBasis-auto _boxSizing-border-box _position-relative _minHeight-0px _minWidth-0px _flexShrink-0 _flexDirection-column"
        >
          <div
            class="_display-flex _flexBasis-auto _boxSizing-border-box _position-relative _minHeight-0px _minWidth-0px _flexShrink-0 _flexDirection-row _gap-t-space-gap52 _alignItems-center"
          >
            <div
              class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
              style="height: 16px; width: 16px; border-top-left-radius: var(--t-radius-roundedFull); border-top-right-radius: var(--t-radius-roundedFull); border-bottom-right-radius: var(--t-radius-roundedFull); border-bottom-left-radius: var(--t-radius-roundedFull);"
            >
              <div
                class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
                style="background-image: url(https://raw.githubusercontent.com/Uniswap/assets/master/blockchains/ethereum/assets/0x6B175474E89094C44Da98b954EedeAC495271d0F/logo.png);"
              />
              <img
                alt=""
                class="css-accessibilityImage-9pa8cd"
                draggable="false"
                src="https://raw.githubusercontent.com/Uniswap/assets/master/blockchains/ethereum/assets/0x6B175474E89094C44Da98b954EedeAC495271d0F/logo.png"
              />
            </div>
            <span
              class="font_subHeading _display-inline _boxSizing-border-box _whiteSpace-pre-wrap _mt-0px _mr-0px _mb-0px _ml-0px _color-neutral1 _fontFamily-f-family _wordWrap-break-word _fontSize-f-size-smal108 _lineHeight-f-lineHeigh1269072494 _fontWeight-f-weight-bo3548"
              data-disable-theme="true"
            >
              252.074
               
              DAI
            </span>
            <svg
              fill="none"
              height="12px"
              stroke="#131313"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="12px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <line
                x1="5"
                x2="19"
                y1="12"
                y2="12"
              />
              <polyline
                points="12 5 19 12 12 19"
              />
            </svg>
            <div
              class="css-view-175oi2r r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
              style="height: 16px; width: 16px; border-top-left-radius: var(--t-radius-roundedFull); border-top-right-radius: var(--t-radius-roundedFull); border-bottom-right-radius: var(--t-radius-roundedFull); border-bottom-left-radius: var(--t-radius-roundedFull);"
            >
              <div
                class="css-view-175oi2r r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw r-backgroundSize-4gszlv"
                style="background-image: url(https://raw.githubusercontent.com/Uniswap/assets/master/blockchains/ethereum/assets/0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2/logo.png);"
              />
              <img
                alt=""
                class="css-accessibilityImage-9pa8cd"
                draggable="false"
                src="https://raw.githubusercontent.com/Uniswap/assets/master/blockchains/ethereum/assets/0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2/logo.png"
              />
            </div>
            <span
              class="font_subHeading _display-inline _boxSizing-border-box _whiteSpace-pre-wrap _mt-0px _mr-0px _mb-0px _ml-0px _color-neutral1 _fontFamily-f-family _wordWrap-break-word _fontSize-f-size-smal108 _lineHeight-f-lineHeigh1269072494 _fontWeight-f-weight-bo3548"
              data-disable-theme="true"
            >
              0.10684
               
              WETH
            </span>
          </div>
          <span
            class="font_body _display-inline _boxSizing-border-box _whiteSpace-pre-wrap _mt-0px _mr-0px _mb-0px _ml-0px _color-neutral1 _fontFamily-f-family _wordWrap-break-word _fontSize-f-size-smal108 _lineHeight-f-lineHeigh1269072494 _fontWeight-f-weight-bo3548"
            data-disable-theme="true"
          >
            when 0.00042 WETH/DAI
          </span>
        </div>
      </div>
    </div>
    <div
      class="_display-flex _flexBasis-auto _boxSizing-border-box _position-relative _minHeight-0px _minWidth-0px _flexShrink-0 _flexDirection-column _alignItems-center _borderTopColor-transparent _borderRightColor-transparent _borderBottomColor-transparent _borderLeftColor-transparent _borderTopLeftRadius-t-radius-ro840690004 _borderTopRightRadius-t-radius-ro840690004 _borderBottomRightRadius-t-radius-ro840690004 _borderBottomLeftRadius-t-radius-ro840690004 _borderTopWidth-t-space-spa94665586 _borderRightWidth-t-space-spa94665586 _borderBottomWidth-t-space-spa94665586 _borderLeftWidth-t-space-spa94665586 _height-23px _justifyContent-center _width-23px _borderBottomStyle-solid _borderTopStyle-solid _borderLeftStyle-solid _borderRightStyle-solid"
    >
      <button
        aria-checked="false"
        class="is_Checkbox _borderTopColor-0disabled-neutral3 _borderRightColor-0disabled-neutral3 _borderBottomColor-0disabled-neutral3 _borderLeftColor-0disabled-neutral3 _borderTopColor-0hover-neutral2 _borderRightColor-0hover-neutral2 _borderBottomColor-0hover-neutral2 _borderLeftColor-0hover-neutral2 _display-flex _flexBasis-auto _boxSizing-border-box _position-relative _minHeight-0px _minWidth-0px _flexShrink-0 _flexDirection-column _borderTopLeftRadius-t-radius-ro840690006 _borderTopRightRadius-t-radius-ro840690006 _borderBottomRightRadius-t-radius-ro840690006 _borderBottomLeftRadius-t-radius-ro840690006 _opacity-0 _alignItems-center _backgroundColor-transparent _borderTopColor-neutral2 _borderRightColor-neutral2 _borderBottomColor-neutral2 _borderLeftColor-neutral2 _borderTopWidth-t-space-spa94665587 _borderRightWidth-t-space-spa94665587 _borderBottomWidth-t-space-spa94665587 _borderLeftWidth-t-space-spa94665587 _cursor-pointer _height-18px _justifyContent-center _pointerEvents-auto _width-18px _borderTopStyle-solid _borderRightStyle-solid _borderBottomStyle-solid _borderLeftStyle-solid"
        data-disable-theme="true"
        data-state="unchecked"
        role="checkbox"
        type="button"
        value="on"
      />
    </div>
  </div>
  <div
    style="display: contents;"
  />
</span>
`;
